src/blueapi/service/main.py

```diff
@@ -74,6 +74,15 @@ def submit_task(
     return TaskResponse(task_name=name)
 
 
+@app.get("/status")
+async def get_status(handler: Handler = Depends(get_handler)) -> WorkerState:
+    return (
+        handler.worker.state
+        if isinstance(handler.worker, RunEngineWorker)
+        else WorkerState.UNKNOWN
```

Comment:
You can do work in properties as well. Having a `state` attribute that may be either a direct field access or a property is arguably the better, less java-y way of doing it.

Reply:
Of those two I'd rather property so as to keep it read-only

Reply:
Since the state that is being returned is actually retrieving the most recently stored value set by the RunEngine, I think keeping it as a method is the best option. If the state expands and requires calculating e.g. the state of multiple submitted plans, having it as a method signature that implies work is being done and not just accessing a field seems sensible.
Tangential thought: do we want to store when the State was last changed? 
Would allow get /worker/state = {"state": "stopping", "since": "2 hours ago"} which passes the eye test for "something is wrong here". Maybe only for transient states?

Reply:
>Since the state that is being returned is actually retrieving the most recently stored value set by the RunEngine, I think keeping it as a method is the best option

I'm not sure I follow the logic here. The fact there is a RunEngine involved should be irrelevant to the end user and the state is 'just' an attribute of the service.

>Tangential thought: do we want to store when the State was last changed?

I like this idea, although a timestamp instead of a relative string would be more usable. No reason not to keep it consistent for all statuses as well. Could also let the history be queried in future - `GET /worker/state?since=20230522_120000`.

Reply:
I'll make a ticket for supporting /state?since= queries, presumably returning a list of {status, timestamp} objects? Seems that task_id for state in {running, pausing...} would be useful info to store too.

Reply:
> If the state expands and requires calculating e.g. the state of multiple submitted plans

I think I'm fairly keen that it doesn't get complicated. I'm reasonably sold on the property argument.

Reply:
...but not hugely bothered either way so happy to just approve when CI is fixed